pandas/io/tests/test_parsers.py

```diff
@@ -2664,6 +2664,40 @@ def test_empty_header_read(count):
         for count in range(1, 101):
             test_empty_header_read(count)
 
+    def test_uneven_lines_with_usecols(self):
+        # See gh-12203
+        csv = r"""a,b,c
+        0,1,2
+        3,4,5,6,7
+        8,9,10
+        """
+
+        # make sure that an error is still thrown
+        # when the 'usecols' parameter is not provided
+        name = self.__class__.__name__
+        msg = "Expected \d+ fields in line \d+, saw \d+"
```

Comment:
you wont' be able to remove it (well *shouldn't* as its in the public API). But it inherites from ``Exception``. if you can make it inherit from ``ValueError`` then I think we'd take that. See what happens. lmk.